### PR TITLE
Reprogramming the brain

### PR DESCRIPTION
**Pull request by jonmagic**
I installed the ArduinoNano we have right on the large breadboard and moved the 1-wire to rx/tx circuit to the board as well so everything is wired up and easy to plug into and start iterating on. Because of this change I had to update platformio.ini to use the correct board setting.

In brain.ino we completely rewrote the parsing logic to be more similar to script/saleae and also added a method to receive a message. Currently the brain is just sending the seconds number 3 over and over again in `loop`, but you can see in the commented out code in the `setup` method the direction I was going with things.

While debugging things we made some more Saleae recordings:
- fake-insert: us trying to impersonate the cd changer when a cd changer is inserted
- fake-insert2: same thing different attempt
- handshake-on-startup-without-changer: the one sided conversation of the stereo when powered on with no cd changer attached
- handshake-on-startup: the two sided conversation of the stereo and cd changer when powered on

Finally I updated script/saleae because I was seeing some wider ranges in pulse width.

/cc @jch @patrickmckenna

### CODE REVIEW

**Comment by jonmagic** [comment]
parsed handshake-on-startup.csv

``` json
[
  {
    "bits": "1101010100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 62.635083333334
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.329166666665998
  },
  {
    "bits": "111001101111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 5.330749999999996
  },
  {
    "bits": "1100001011111111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 246.21066666666698
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 4.0762500000000035
  },
  {
    "bits": "1100001101001000000100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "1000000100",
    "milliseconds_to_next_message": 4.038166666667009
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 7.520583333332942
  },
  {
    "bits": "1100001101001000000100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "1000000100",
    "milliseconds_to_next_message": 30.917166666666994
  },
  {
    "bits": "111001110100010010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 7.068500000000033
  },
  {
    "bits": "1100001101001000000100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "1000000100",
    "milliseconds_to_next_message": 50.88308333333402
  },
  {
    "bits": "110000110100100010",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "100010",
    "milliseconds_to_next_message": 90.61499999999988
  },
  {
    "bits": "111001101111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 7.320749999999876
  },
  {
    "bits": "111001110100010010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 335.79691666666696
  },
  {
    "bits": "11000011010000000000000101",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "00000000000101",
    "milliseconds_to_next_message": 514.1220833333329
  },
  {
    "bits": "1100001011111110000100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 260.6416666666669
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.884083333333013
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 7.325083333332927
  },
  {
    "bits": "111001101111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 3.517666666665864
  },
  {
    "bits": "110000110100100010",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "100010",
    "milliseconds_to_next_message": 0
  }
]
```

**Comment by jonmagic** [comment]
parsed handshake-on-startup-without-changer.csv

``` json
[
  {
    "bits": "1101010100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 62.63158333333302
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.32608333333401
  },
  {
    "bits": "111001101111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 5.32775000000002
  },
  {
    "bits": "1100001011111111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 246.21099999999998
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 16.82708333333305
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 161.580583333334
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 4.88566666666701
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 168.56325000000007
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 4.889333333333079
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 207.33800000000002
  },
  {
    "bits": "1101010100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 62.62908333333295
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.324166666666908
  },
  {
    "bits": "111001101111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 5.327833333332865
  },
  {
    "bits": "1100001011111111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 244.22349999999992
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 12.85333333333294
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 153.62050000000104
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 13.843583333333909
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 151.64350000000002
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 14.84749999999968
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 162.58541666666605
  },
  {
    "bits": "111001110100010010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 1121.604333333333
  },
  {
    "bits": "1100001011111110000100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 236.81841666666736
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 8.86666666666569
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 9.311166666666093
  },
  {
    "bits": "111001101111",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 130.7379166666669
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 8.864999999999679
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 155.63041666666598
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 8.868500000000168
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 0
  }
]
```

File: platformio.ini [hunk 1]
```diff
@@ -20,6 +20,6 @@
 [env:uno]
 platform = atmelavr
 framework = arduino
-board = uno
-# board = nanoatmega328
+# board = uno
```
Comment (jch) [inline platformio.ini:23]: Thanks for leaving this commented out so it's easy to switch if we need to.

File: src/brain.ino [hunk 2]
```diff
@@ -1,215 +1,114 @@
...
-  //   start = 0;
-  //   end = 0;
   // }
 }
+
+void loop() {
+  char three[] = "10100011010010010110";
+  send_message(three);
+  delay(1000);
+}
```
Comment (jch) [inline src/brain.ino:114]: Nice. This makes it nice and easy to understand.

**Comment by jch** [comment]
Looks good. Bummed I missed out last night. Merge when ready :+1:

**Pull request merged**